sklearnex/covariance/incremental_covariance.py

```diff
@@ -293,21 +312,34 @@ def _onedal_fit(self, X, queue=None):
         return self
 
     # expose sklearnex pairwise_distances if mahalanobis distance eventually supported
-    @wrap_output_data
     def mahalanobis(self, X):
         if sklearn_check_version("1.0"):
-            self._validate_data(X, reset=False, copy=self.copy)
-        else:
-            check_array(X, copy=self.copy)
+            self._check_feature_names(X, reset=False)
 
+        xp, _ = get_namespace(X)
         precision = self.get_precision()
-        with config_context(assume_finite=True):
-            # compute mahalanobis distances
-            dist = pairwise_distances(
-                X, self.location_[np.newaxis, :], metric="mahalanobis", VI=precision
-            )
+        # compute mahalanobis distances
+        # pairwise_distances will check n_features (via n_feature matching with
+        # self.location_) , and will check for finiteness via check array
+        # check_feature_names will match _validate_data functionally
+        location = self.location_[np.newaxis, :]
+        if "numpy" not in str(xp).lower():
```

Comment:
Could be replaced by https://github.com/samir-nasibli/scikit-learn-intelex/blob/4179f22b5671b9ff83baa4d2588556e791459c58/onedal/_device_offload.py#L210
But again it make sense to do after #1861 or #1813